dotcom-rendering/src/lib/identity.ts

```diff
@@ -22,3 +28,159 @@ export async function isSignedInWithOktaAuthState(): Promise<
 			};
 		});
 }
+
+type OktaAuthState = IdentityAuthState<never, CustomIdTokenClaims>;
+
+type SignedOutWithCookies = { kind: 'SignedOutWithCookies' };
+export type SignedInWithCookies = { kind: 'SignedInWithCookies' };
+type SignedOutWithOkta = { kind: 'SignedOutWithOkta' };
+export type SignedInWithOkta = {
+	kind: 'SignedInWithOkta';
+	accessToken: AccessToken<never>;
+	idToken: IDToken<CustomIdTokenClaims>;
+};
+
+export type AuthStatus =
+	| { kind: 'Pending' }
+	| SignedOutWithCookies
+	| SignedInWithCookies
+	| SignedOutWithOkta
+	| SignedInWithOkta;
+
+// relates to getUserFromCookie() below but not used in user-features - however may be useful for other purposes
+type IdentityUserFromCache = {
+	dates: { accountCreatedDate: string };
+	publicFields: {
+		displayName: string;
+	};
+	statusFields: {
+		userEmailValidated: boolean;
+	};
+	id: number;
+	rawResponse: string;
+} | null;
+
+let userFromCookieCache: IdentityUserFromCache = null;
+const cookieName = 'GU_U';
+const getUserCookie = (): string | null => getCookie({ name: cookieName });
+const decodeBase64 = (str: string): string =>
+	decodeURIComponent(
+		escape(
+			window.atob(
+				str.replace(/-/g, '+').replace(/_/g, '/').replace(/,/g, '='),
+			),
+		),
+	);
+
+const getUserFromCookie = (): IdentityUserFromCache => {
+	if (userFromCookieCache === null) {
+		const cookieData = getUserCookie();
+
+		if (cookieData) {
+			const userData = JSON.parse(
+				decodeBase64(cookieData.split('.')[0] ?? ''),
+			) as string[];
```

Comment:
Non-blocking: this can cause the app to crash if the cookie value is malformed!

Reply:
Would perhaps error handling in a try/catch help?